### PR TITLE
Add mplcursors to third-party packages.

### PR DESCRIPTION
**Pull request by anntzer**
Self-plug :-)

@joferkington I would like to add a paragraph comparing the approaches of mpldatacursor and mplcursors, but I'd like to make sure that you agree with it.  My feeling that mpldatacursor tries to provide as many customizations to the annotation as possible as kwargs in the call to `datacursor()`, whereas I tried to make `mplcursors` as minimal as possible, by letting instead the user hook into annotation-adding or removing events to customize the annotation as he sees fit.

If we can agree on a comparison, then perhaps the two sections can be merged together, too.

### CODE REVIEW

File: doc/thirdpartypackages/index.rst [hunk 1]
```diff
@@ -134,6 +134,15 @@ toolkit written by Joe Kington to provide interactive "data cursors"
 (clickable annotation boxes) for matplotlib.
 
 
+.. _toolkit_mplcursors:
+
+mplcursors
+==========
+
+`mplcursors <https://github.com/anntzer/mplcursors>` provides interactive
```
Comment (QuLogic) [inline doc/thirdpartypackages/index.rst]: The trailing underscores are missing to turn this into a link. Also, why not link to RTD instead of GitHub?

Comment (anntzer) [inline doc/thirdpartypackages/index.rst]: Thanks for the suggestion, fixed.

File: doc/thirdpartypackages/index.rst [hunk 2]
```diff
@@ -134,6 +134,15 @@ toolkit written by Joe Kington to provide interactive "data cursors"
 (clickable annotation boxes) for matplotlib.
 
 
+.. _toolkit_mplcursors:
+
+mplcursors
+==========
+
+`mplcursors <https://mplcursors.readthedocs.io>_` provides interactive
```
Comment (QuLogic) [inline doc/thirdpartypackages/index.rst]: The underscore goes outside the backtick.

Comment (anntzer) [inline doc/thirdpartypackages/index.rst]: fixed again.

**Comment by tacaswell** [comment]
@anntzer Do you want to merge this as-is or wait for feedback from @joferkington ?

**Comment by joferkington** [comment]
Looks good to me! Greek free to merge the sections as well, it you'd like.

Also, my apologies for being so out of the loop on things x lately.

**Comment by tacaswell** [comment]
backported to v1.5.3-doc as d134631

**Pull request merged**